src/Security/Authentication/Negotiate/src/Internal/LdapAdapter.cs

```diff
@@ -129,12 +202,145 @@ private static void GetNestedGroups(LdapConnection connection, ClaimsIdentity pr
                     if (processedGroups.Contains(nestedGroupDN))
```

Comment:
Is this still necessary given we have the same check at the top of the method?

Reply:
As far as I can tell you are right, it would not be necessary. Currently this would only avoid another recursion including one more LDAP search request before returning back to this loop, but it could be removed.